File: 1505-create-target-array-in-the-given-order/solution.java
class Solution {
    public int[] createTargetArray(int[] nums, int[] index) {
        int n = nums.length;
        int[] res = new int[n];
        for (int i = 0; i < n; i++) {
            int indexNth = index[i];
            for (int j = n - 1; j > indexNth; j--) {
                res[j] = res[j - 1];
            }
            res[indexNth] = nums[i];
        }

        return res;
    }
}

File: 1505-create-target-array-in-the-given-order/solution.go
func createTargetArray(nums []int, index []int) []int {
    target := make([]int, 0)
    for i := 0; i < len(nums); i++ {
        idx := index[i]
        val := nums[i]

        // [0, 1, 2, 3, 4]
        if len(target) == 0 || idx == len(target) {
            target = append(target, val)
        } else {
            target = append(target[:idx + 1], target[idx:]...)
            target[idx] = val
        }
    }

    return target
}
